auv_sim/auv_sim_bringup/launch/inc/start_world.launch.xml

```diff
@@ -16,8 +15,9 @@
     <arg name="verbose" value="true" />
   </include>
 
-  <node name="publish_world_models" pkg="uuv_assistants" type="publish_world_models.py"
-    output="screen">
-    <rosparam command="load" file="$(find auv_sim_bringup)/config/world_models.yaml" />
-  </node>
-</launch>
\ No newline at end of file
+  <group if="$(eval arg('world_name') == arg('seabed_world'))">
+    <node name="publish_world_models" pkg="uuv_assistants" type="publish_world_models.py" output="screen">
+      <rosparam command="load" file="$(find auv_sim_bringup)/config/world_models.yaml" />
+    </node>
+  </group>
+</launch>
```

Comment:
The publish_world_models node is only relevant for the seabed world and is unnecessary for the pool environment. With this conditional check, the node will only execute when world_name is set to seabed_world

Reply:
I understand what you want to achieve here. But what if we have another world to add? Will we make this launch file have bunch of if else sections? Maybe we can add 

launch/inc/worlds/spawn_pool.launch
...

launch/inc/worlds/spawn_<world_name>.launch


Then we can really do, 

roslaunch auv_sim_bringup start.launch world:=seabed

Which will 

<include file="..... launch/inc/worlds/spawn_$(arg world).launch">

And in that folder we add all spawn_... for each world we want to use. 

I also think that this node you're referring can be included inside the world file just like the pool world, but I think editing world files and not being able to benefit from launch files' arg,find,if,else functions will make things harder eventually, thus keeping world specific launch file could be a good idea. 
 
Of course this could be investigated, no pressure. Maybe you just shake the tree and see what comes down?

Reply:
I've made some updates about this. You can check the other PR @melihokur17 started.